examples/cvae_dim_example/README.md

```diff
@@ -0,0 +1,51 @@
+# Federated Dimensionality reduction using CVAEs Example
+In this example, clients first reduce the dimensionality of the data using an already federally trained conditional variational auto-encoder (please refer to the CVAE example). Then, the data in the reduced dimension is used to train a small neural network model (`mnist_model.py`). Clients in this example inherit from a `CVAETrainer` class to use the CVAE processing functionalities i.e. set the client-specific condition and reduce the dimensionality of data via CVAE data pre-processor. Federated training for the main task of MNIST classification can be done with any of the FL approaches by specifying the parent client class. The default parent client class in this example is the `BasicClient` and the server uses Federated Averaging.
+
+## Running the Example
+In order to run the example, first ensure you have the virtual env of your choice activated and run
+```
+pip install --upgrade pip
+pip install -r requirements.txt
+```
+to install all of the dependencies for this project.
+
+## Starting Server
+
+The next step is to start the server by running
+```
+python -m examples.cvae_dim_example.server  --config_path /path/to/config.yaml
+```
+from the FL4Health directory. The following arguments must be present in the specified config file:
+* `n_clients`: number of clients the server waits for in order to run the FL training
+* `local_epochs`: number of epochs each client will train for locally
+* `batch_size`: size of the batches each client will train on
+* `n_server_rounds`: The number of rounds to run FL
+* `checkpoint_path`: path to save the best server model
+* `latent_dim`: size of the latent vector in the CVAE or VAE model
+* `CVAE_model_path`: path to the saved CVAE model for dimesionality reduction
```

Comment:
That totally makes sense. However, we need to somehow make sure they are loading a personalized model that is not considerably different from other client's models. Otherwise, the mappings might end up in different latent spaces (increased heterogeneity). So, I did this on purpose to emphasize this fact. Users can still load different CVAEs into their clients if they want but they should be careful about that. I am going to make a note in the readme file about this.

Reply:
I see, yeah that makes sense.